src/app/Main/Body/QueryPage/Drawer/Displayed_Features/DisplFeatures-Selector.component.tsx

```diff
@@ -58,29 +57,44 @@ export function DisplayedFeatureSelectorComponent(
         props.mainFilterAttributes,
         [props.otherFeature]
     );
-    const [isNotSelect, selectedValues]: [
-        boolean,
-        FilterType[]
-    ] = generateFeatureList(props.activeFeature);
+
+    const isNotSelect: boolean = _.isEmpty(props.activeFeature);
+    const selectedValue: FilterType | null = isNotSelect
+        ? null
+        : props.activeFeature;
 
     const dropDownValuesObj: {
         value: string;
         label: string;
     }[] = generateTranslatedSelectorObject(offeredAttributes, t);
-    const selectedValuesObj: {
-        value: string;
-        label: string;
-    }[] = generateTranslatedSelectorObject(selectedValues, t);
+
+    const selectedValueObj:
+        | {
+              value: string;
+              label: string;
+          }[]
+        | [] =
+        selectedValue === null
+            ? []
+            : [
+                  {
+                      value: selectedValue,
+                      label: t(`Filters.${selectedValue}`),
+                  },
+              ];
 
     const handleChange = (
         selectedOption: ValueType<{ value: string; label: string }>,
         keyName: FilterType | FeatureType
     ): void => {
         if (selectedOption !== undefined && !Array.isArray(selectedOption)) {
-            props.onChange(
-                selectedOption as { value: string; label: string } | null,
-                keyName
-            );
+            const option = selectedOption as { value: string; label: string };
+            if (option.value !== selectedValue) {
+                props.onChange(
+                    selectedOption as { value: string; label: string } | null,

```

Comment:
Reject: If you say selectedOption can be null here you say also the above if statement could fail with an exception. So to indicate that you know what you do, you can just use option here as you converted it to the suspected type above.
If option can indeed be null you have to handle this case separately.

Reply:
Done.